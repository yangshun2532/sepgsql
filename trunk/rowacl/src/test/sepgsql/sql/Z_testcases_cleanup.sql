--
-- Initial setup of SE-PostgreSQL testcases
--
-- selinux_user: unconfined_u
-- selinux_role: unconfined_r
-- selinux_type: unconfined_t
-- selinux_range: s0-s0:c0.c15

-- -------------------------------------------------------
-- Test B : Row-Level Access Control
-- -------------------------------------------------------

-- memo: to confirm default security context on INSERT
SELECT security_context, * FROM b1;
SELECT security_context, * FROM b2;

DROP TABLE b1;
DROP TABLE b2;

-- -------------------------------------------------------
-- Test C : Column-Level Access Control
-- -------------------------------------------------------

DROP TABLE c1;
DROP TABLE c2;

-- -------------------------------------------------------
-- Test D : PK/FK constraint
-- -------------------------------------------------------

DROP TABLE d2;
DROP TABLE d1;

-- -------------------------------------------------------
-- Test E : Trusted Procedure
-- -------------------------------------------------------

DROP TABLE e1;
DROP FUNCTION e2(integer);
DROP FUNCTION e3();

-- -------------------------------------------------------
-- Test F : Extended SQL Grammer
-- -------------------------------------------------------


-- -------------------------------------------------------
-- Test G : Large Object
-- -------------------------------------------------------

