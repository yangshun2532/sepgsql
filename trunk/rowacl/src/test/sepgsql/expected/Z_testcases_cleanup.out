--
-- Initial setup of SE-PostgreSQL testcases
--
-- selinux_user: unconfined_u
-- selinux_role: unconfined_r
-- selinux_type: unconfined_t
-- selinux_range: s0-s0:c0.c15
-- -------------------------------------------------------
-- Test B : Row-Level Access Control
-- -------------------------------------------------------
-- memo: to confirm default security context on INSERT
SELECT security_context, * FROM b1;
                 security_context                  | x  |       y       
---------------------------------------------------+----+---------------
 unconfined_u:object_r:sepgsql_table_t:s0:c1       |  4 | coffee
 unconfined_u:object_r:sepgsql_fixed_table_t:s0    |  2 | coke
 unconfined_u:object_r:sepgsql_fixed_table_t:s0:c1 |  3 | juice
 unconfined_u:object_r:sepgsql_table_t:s0          |  1 | water_updated
 unconfined_u:object_r:sepgsql_table_t:s0          |  5 | beer_updated
 unconfined_u:object_r:sepgsql_table_t:s0          |  6 | wine_updated
 unconfined_u:object_r:sepgsql_table_t:s0:c0       | 10 | soda
(7 rows)

SELECT security_context, * FROM b2;
                security_context                | a  |   b    
------------------------------------------------+----+--------
 unconfined_u:object_r:sepgsql_table_t:s0:c1    |  4 | yellow
 unconfined_u:object_r:sepgsql_ro_table_t:s0:c1 |  5 | black
 unconfined_u:object_r:sepgsql_ro_table_t:s0    |  6 | white
 unconfined_u:object_r:sepgsql_table_t:s0:c0    | 10 | orange
(4 rows)

DROP TABLE b1;
DROP TABLE b2;
-- -------------------------------------------------------
-- Test C : Column-Level Access Control
-- -------------------------------------------------------
DROP TABLE c1;
DROP TABLE c2;
-- -------------------------------------------------------
-- Test D : PK/FK constraint
-- -------------------------------------------------------
DROP TABLE d2;
DROP TABLE d1;
-- -------------------------------------------------------
-- Test E : Trusted Procedure
-- -------------------------------------------------------
DROP TABLE e1;
DROP FUNCTION e2(integer);
DROP FUNCTION e3();
-- -------------------------------------------------------
-- Test F : Extended SQL Grammer
-- -------------------------------------------------------
-- -------------------------------------------------------
-- Test G : Large Object
-- -------------------------------------------------------
